concordium-consensus/src/Concordium/GlobalState/Persistent/Trie.hs

```diff
@@ -193,16 +270,18 @@ instance Functor (Trie k) where
 -- (as for 'Trie'), but also correspond to creating and traversing disk references.
 
 -- |Retrieve the value (if any) corresponding to a given key.
+-- TODO: With bytestring-0.11.3.0 and later, the operations on 'SBS.ShortByteString' can
+-- be used to implement this more efficiently without unpacking.
 lookupF :: (MRecursive m t, Base t ~ TrieF k v, FixedTrieKey k) => k -> t -> m (Maybe v)
 lookupF k = lu (unpackKey k) <=< mproject
     where
         lu [] (Tip v) = pure (Just v)
         lu _ (Tip _) = pure Nothing
-        lu key (Stem pref r) = case stripPrefix pref key of
+        lu key (Stem pref r) = case stripPrefix (SBS.unpack pref) key of
```

Comment:
We could implement a probably more efficient version now by modifying followStem to take a bytestring and a cursor.
It's not much more complicated than comparing two lists.

Reply:
I looked into this. I found that it could save less than 100ns per look-up. That's not completely insignificant in the lookup benchmarks, but in the broader scheme of things it probably isn't significant. It was also not so trivial to implement, at least in my approach, and I think the added complexity of the change doesn't justify it.

Reply:
Ok.